dlt/load/utils.py

```diff
@@ -216,3 +218,44 @@ def _extend_tables_with_table_chain(
                 continue
             result.add(chain_table_name)
     return result
+
+
+def filter_new_jobs(
+    file_names: Sequence[str],
+    capabilities: DestinationCapabilitiesContext,
+    config: LoaderConfiguration,
+) -> Sequence[str]:
+    """Filters the list of new jobs to adhere to max_workers and parallellism strategy"""
+
+    # nothing to do
+    if not file_names:
+        return file_names
+
+    # destination can overwrite ps
+    parallelism_strategy = capabilities.loader_parallelism_strategy or config.parallelism_strategy
+
+    # we only always process one
+    if parallelism_strategy == "sequential":
+        return file_names[:1]
+
+    # find real max workers value
+    max_workers = config.workers
+    if mp := capabilities.max_parallel_load_jobs:
+        max_workers = min(max_workers, mp)
+
+    # regular sequential will take max worker amount no matter the table
+    if parallelism_strategy == "parallel":
+        return file_names[:max_workers]
+
+    # we must ensure there only is one job per table
+    if parallelism_strategy == "table_sequential":
+        filtered_jobs: List[str] = []
```

Comment:
this looks good. but you could also use `group_by` by table name, maybe the code will be simpler?

Reply:
i made a one-liner out of it, not sure if it is easier to read now, but it is more pythonic i guess :)

Reply:
Turns out you need to sort by the key you group by for the groupby to work as expected..